### DISCUSSION TITLE
Mesh bed levelling

### DISCUSSION THREAD

**Discussion by lukysy** [dz0ny/klipper-prusa-mk3s#10] [Q&A] [answered]
How to set more than 3x3 point mesh bed leveling? When trying to set any more detailed mesh bed leveling, the printer does make a mesh network successfully, but during leveling it behaves strangely in some places and moves in all directions.

Any pointers on how to figure out where the moves are defined? I play with the offset and size of the printing area, the number of leveling points, and also with the setting of the screw position. Can they affect the magnets in the heating bed?

Leveling on 3x3 is completely insufficient when printing larger areas.

**Comment by dz0ny** [comment, marked as answer]
The moves are calculated by Klipper, what is defined are magnets as faulty regions, see https://github.com/dz0ny/klipper-prusa-mk3s/blob/main/mk3s/einsy-rambo.cfg#L90 and https://www.klipper3d.org/Bed_Mesh.html#faulty-regions.

> Leveling on 3x3 is completely insufficient when printing larger areas.

Agreed, that is why I left option to specify custom in https://github.com/dz0ny/klipper-prusa-mk3s/blob/main/printer.template.cfg#L79

I would also suggest you try the https://gist.github.com/ChipCE/95fdbd3c2f3a064397f9610f915f7d02?permalink_comment_id=4206084. It's muchs smarter bed leveling as it's based on print object size.
